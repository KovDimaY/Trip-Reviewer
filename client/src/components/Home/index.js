import React from 'react';

import HomeContainer from './../../containers/Home/index';

const Home = () => {
    return (
        <div>
            <HomeContainer />
        </div>
    );
};

export default Home;