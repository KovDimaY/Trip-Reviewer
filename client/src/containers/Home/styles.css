.home-container .loadmore {
  background: #00BCD4;
  text-align: center;
  text-transform: uppercase;
  color: #ffffff;
  font-size: 16px;
  padding: 20px;
  font-weight: 500;
  cursor: pointer;
}
.home-container .empty-view-wrapper {
  padding: 75px 0;
}
.home-container .loadmore:hover {
  background: #009EB3;
  font-size: 18px;
}

.home-container .loadmore:active {
  background: #0495A8;
  font-size: 17px;
}