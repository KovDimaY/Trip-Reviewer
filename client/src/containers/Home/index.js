import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import TripItem from '../../components/TripItem/index';
import { getTrips } from '../../actions';

import './styles.css';

const ITEMS_TO_LOAD = 3;

class HomeContainer extends Component {
  state = {
    showLoadmore: true,
  };

  componentWillMount() {
    const itemsToLoad = 2 * ITEMS_TO_LOAD;
    const startingFrom = 0;
    const order = 'desc';

    this.props.dispatch(getTrips(itemsToLoad, startingFrom, order));
  }

  componentWillReceiveProps(nextProps) {
    const { newTripsCount } = nextProps.trips;

    if (newTripsCount < ITEMS_TO_LOAD) {
      this.setState({ showLoadmore: false });
    }
  }

  loadmore = () => {
    const { list } = this.props.trips;
    const newItemsToLoad = ITEMS_TO_LOAD;
    const startingFrom = list.length;
    const order = 'desc';

    this.props.dispatch(getTrips(newItemsToLoad, startingFrom, order, list));
  }

  renderItems() {
    const { list = [] } = this.props.trips;

    if (list.length < 1) {
      return (
        <div className="empty-view">
          <p>There is no any review yet.</p>
          <p>Be the first who posts one! :D</p>
        </div>
      );
    }

    return list.map(item => <TripItem {...item} key={item._id} />);
  }

  renderLoadMoreButton() {
    if (this.state.showLoadmore) {
      return (
        <div className="loadmore" onClick={this.loadmore}>
          Load More
        </div>
      );
    }

    return null;
  }

  render() {
    return (
      <div className="home-container limited-width-shadow">
        {this.renderItems(this.props.trips)}
        {this.renderLoadMoreButton()}
      </div>
    );
  }
}

HomeContainer.propTypes = {
  trips: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  trips: state.trips,
});

export default connect(mapStateToProps)(HomeContainer);
