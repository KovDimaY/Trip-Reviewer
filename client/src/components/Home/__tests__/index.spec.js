import React from 'react';
import { create } from 'react-test-renderer';
 
import Home from './../../Home';

jest.mock('./../../../containers/Home/index', () => 'HomeContainer');

const mockComponent = props => {
    return (
        <Home {...props} />
    );
};
 
describe('<Home />', () => {
    it('should render component', () => {
        const props = {
            user: {
                login: {
                    name: 'name',
                    lastname: 'lastname',
                    email: 'email'
                }
            }
        };
        const tree = create(mockComponent(props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});