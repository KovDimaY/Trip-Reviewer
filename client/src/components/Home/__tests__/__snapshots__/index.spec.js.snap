// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render component 1`] = `
<div>
  <HomeContainer />
</div>
`;
